Add plugin: Auto Bullet

takitsuba opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/takitsuba/obsidian-auto-bullet

## Release Checklist
- [x] I have tested the plugin on
  - [ ]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [x]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [ ] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

Zachatoo commented:
[# Obsidian Auto Bullet](https://github.com/takitsuba/obsidian-auto-bullet/blob/a66ad2f8125c4b6b2e8b29e47f6986bf368a7ba3/README.md#L1)
Naming something "Obsidian xyz" is reserved for first party products we create, please change this header.

["fundingUrl": "",](https://github.com/takitsuba/obsidian-auto-bullet/blob/a66ad2f8125c4b6b2e8b29e47f6986bf368a7ba3/manifest.json#L9)
This is meant for links to services like _Buy me a coffee_, _GitHub sponsors_ and so on, if you don't have such a link remove this line.

["hasSettings": true](https://github.com/takitsuba/obsidian-auto-bullet/blob/a66ad2f8125c4b6b2e8b29e47f6986bf368a7ba3/manifest.json#L11)
Do not use properties that are not documented in your plugin's [manifest](https://docs.obsidian.md/Reference/Manifest).

[Copyright (C) 2020-2025 by Dynalist Inc.](https://github.com/takitsuba/obsidian-auto-bullet/blob/a66ad2f8125c4b6b2e8b29e47f6986bf368a7ba3/LICENSE#L1)
Please update this.

[.setName('Half-width Space')](https://github.com/takitsuba/obsidian-auto-bullet/blob/a66ad2f8125c4b6b2e8b29e47f6986bf368a7ba3/main.ts#L90), [.setName('Full-width Space')](https://github.com/takitsuba/obsidian-auto-bullet/blob/a66ad2f8125c4b6b2e8b29e47f6986bf368a7ba3/main.ts#L100)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

---

Optional feedback:

Your plugin also adds bullet points in unexpected situations, such as when in a code block. Consider updating your code to account for code blocks or other markdown elements that usually wouldn't have bullet points in them.

takitsuba commented:
Thank you for your feedback. I've addressed all the issues:
- Changed the plugin name to "Auto Bullet"
- Removed the unused fundingUrl from the manifest
- Removed the undocumented hasSettings property
- Updated the copyright notice
- Fixed UI text to use sentence case

Version 1.0.4 fully addresses the issue with bullet points in code blocks and other Markdown elements.
The plugin now properly handles code blocks, tables, blockquotes, headings, math blocks, and other elements where bullet points should not be automatically inserted.

joethei commented:
[hotkeys: [](https://github.com/takitsuba/obsidian-auto-bullet/blob/1b0a151d338edddaf9d1e6109a519731d2d86457/main.ts#L63-L68)
You should not provide a default hotkey. The reason being that it's easy to pick a hotkey that a user already has configured and they could get confused when that key doesn't do what they expect. Also, it's hard choosing a safe default hotkey that's available for all operating systems & does not conflict with existing plugins.

[this.registerCodeMirror((cm: CodeMirrorEditor) => {](https://github.com/takitsuba/obsidian-auto-bullet/blob/1b0a151d338edddaf9d1e6109a519731d2d86457/main.ts#L72-L104)
This code does nothing, we removed the `registerCodeMirror` function entirely some time ago.
All that's left is an empty shell so that existing plugins don't break.

takitsuba commented:
Thank you for getting back to me.

- I removed the default key configuration
- I removed registerCodeMirror function references
